### PR TITLE
scdoc: adding subsubsection, audio, video, text decoration, some HTML tags, LaTex syntax and MathML syntax 

### PR DESCRIPTION
**Pull request by prko**
## Purpose and Motivation
To make the SCDoc (SuperCollider help documents) more user friendly, I added subsubsections (soft-coded), some HTML tags, LaTex and MathML.

- https://github.com/supercollider/supercollider/issues/6131
- https://scsynth.org/t/scdoc-schelp-style-add-on-latex-mathml-subsubsection-multimedia-file-iframe-sub-sup-small-line-height-hr/9242

## Types of changes

- Documentation
- New feature

## To-do list

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review, but this PR may change dynamically as other users provide feedback.

### CODE REVIEW

**Comment by prko** [comment]
PDF of the documentation:
- [HTML, LaTeX and MathML Syntax in SCDocs _ SuperCollider 3.14.0-dev Help with default css.pdf](https://github.com/supercollider/supercollider/files/14987107/HTML.LaTeX.and.MathML.Syntax.in.SCDocs._.SuperCollider.3.14.0-dev.Help.pdf)
- [HTML, LaTeX and MathML Syntax in SCDocs _ SuperCollider 3.14.0-dev Help with custom css.pdf](https://github.com/supercollider/supercollider/files/14987106/HTML.LaTeX.and.MathML.Syntax.in.SCDocs._.SuperCollider.3.14.0-dev.Help._.custom.css.pdf)

**Review by nhthn — CHANGES_REQUESTED** [review]
I have my doubts over whether SCDoc should be expanded to keep glomming on new features, but more importantly: I'm afraid modifying the generated HTML with tons of `.replace` calls isn't the right way to do this. The correct way to extend SCDoc syntax is to modify the SCDoc parser and HTML generator in the `SCDoc` directory. Unfortunately, as written, this is very fragile and hacky.

File: SCClassLibrary/SCDoc/SCDocRenderer.sc [hunk 1]
```diff
@@ -247,9 +247,11 @@ SCDocHTMLRenderer {
 		<< "</script>\n"
 		<< "<script src='" << baseDir << "/scdoc.js' type='text/javascript'></script>\n"
 		<< "<script src='" << baseDir << "/docmap.js' type='text/javascript'></script>\n" // FIXME: remove?
-		<< "<script src='" << baseDir << "/frontend.js' type='text/javascript'></script>\n"
+		// << "<script src='" << baseDir << "/frontend.js' type='text/javascript'></script>\n" //does not exist
 		// QWebChannel access
-		<< "<script src='qrc:///qtwebchannel/qwebchannel.js' type='text/javascript'></script>\n"
+		// << "<script src='qrc:///qtwebchannel/qwebchannel.js' type='text/javascript'></script>\n" // does not exist
+		<< "<script src='https://polyfill.io/v3/polyfill.min.js?features=es6'></script>\n"
+		<< "<script id='MathJax-script' async src='https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js'></script>\n"
```
Comment (nhthn) [inline SCClassLibrary/SCDoc/SCDocRenderer.sc:254]: Not a good idea -- this makes SCDoc connect to the Internet.

**Comment by prko** [comment]
@nhthn 

Thank you for your valuable feedback!

> I have my doubts over whether SCDoc should be expanded to keep glomming on new features, but more importantly: I'm afraid modifying the generated HTML with tons of `.replace` calls isn't the right way to do this. The correct way to extend SCDoc syntax is to modify the SCDoc parser and HTML generator in the `SCDoc` directory. Unfortunately, as written, this is very fragile and hacky.

- **To:** _I have my doubts over whether SCDoc should be expanded to keep glomming on new features._
  1. With the focus on new features, some of the new features I suggest are not always necessary. I can remove them if one of the reviewers points out the specific feature. 
  2. However, I think your thoughts here are more focused on extending SCDoc, there is no way to implement the features I suggest without extending SCDoc, if I understand its system correctly. So I think the core issue is whether or not these features are needed in SuperCollider's help document browser. 
  3. My position is that they are necessary if the SC help browser is to be easily readable by newcomers. 
  4. Also, some Quarks contain long and complicated documentations, tutorials and guides; I think the new feature proposed here would help to formulate these documents in a reader-friendly and author-friendly way.
- **To:** _I'm afraid modifying the generated HTML with tons of `.replace` calls isn't the right way to do this. Unfortunately, as written, this is very fragile and hacky._
In [288a1c8](https://github.com/supercollider/supercollider/pull/6254/commits/288a1c80bd9ee09793c9ffa5969c17dadddb858c), I reduced the repetition of `.replace` by adding a method.
There is one thing I should mention: the previous version, untouched by me, already has `.replace`:
https://github.com/supercollider/supercollider/blob/28a0b12216fb97e4487373b2cf00a5ec28850f1f/SCClassLibrary/SCDoc/SCDocRenderer.sc#L50-L52 At first glance this appears to be a one-off, but the method is used frequently throughout _SCDocRenderer.sc_.
- **The correct way to extend SCDoc syntax is to modify the SCDoc parser and HTML generator in the `SCDoc` directory.**
- Um... I don't know how to do it... Sorry, but I bet I spent a lot of time (almost two weeks due to the implementation of replaceRegex with two .*. I finally gave up) trying to solve some implementation problems. If you give me more tips on how to do it, I can learn the way and try to achieve it!


PDF of the corresponding documentation (it includes a newly added section):
- [HTML, LaTeX and MathML Syntax in SCDocs _ SuperCollider 3.14.0-dev Help with default css.pdf](https://github.com/supercollider/supercollider/files/15004625/HTML.LaTeX.and.MathML.Syntax.in.SCDocs._.SuperCollider.3.14.0-dev.Help.with.default.css.pdf)
- [HTML, LaTeX and MathML Syntax in SCDocs _ SuperCollider 3.14.0-dev Help with custom css.pdf](https://github.com/supercollider/supercollider/files/15004624/HTML.LaTeX.and.MathML.Syntax.in.SCDocs._.SuperCollider.3.14.0-dev.Help.with.custom.css.pdf)

File: SCClassLibrary/SCDoc/SCDocRenderer.sc (hunk 1 above)
Comment (prko) [reply SCClassLibrary/SCDoc/SCDocRenderer.sc]: Thank you very much! I added files to be able to use LaTeX and MathML offline.
[f129943](https://github.com/supercollider/supercollider/pull/6254/commits/f129943568073ad75baa90c0bb9a888c051e5588)

and changed the code:
https://github.com/supercollider/supercollider/pull/6254/commits/288a1c80bd9ee09793c9ffa5969c17dadddb858c#diff-16b57c89cf950f36adc60ebaf32b5fe84d3b6944a9132f91da1960c65e41a6b6R279-R287

https://github.com/supercollider/supercollider/blob/288a1c80bd9ee09793c9ffa5969c17dadddb858c/SCClassLibrary/SCDoc/SCDocRenderer.sc#L280-L286

**Comment by prko** [comment]
PDF of the corresponding documentation (it includes a newly added section):
- [HTML.LaTeX.and.MathML.Syntax.in.SCDocs._.SuperCollider.3.14.0-dev.Help.with.custom.css.pdf](https://github.com/supercollider/supercollider/files/15009311/HTML.LaTeX.and.MathML.Syntax.in.SCDocs._.SuperCollider.3.14.0-dev.Help.with.custom.css.pdf)
- [HTML.LaTeX.and.MathML.Syntax.in.SCDocs._.SuperCollider.3.14.0-dev.Help.with.default.css.pdf](https://github.com/supercollider/supercollider/files/15009312/HTML.LaTeX.and.MathML.Syntax.in.SCDocs._.SuperCollider.3.14.0-dev.Help.with.default.css.pdf)

**Comment by prko** [comment]
PDF of the corresponding documentation (it includes a newly added section):
- [HTML.LaTeX.and.MathML.Syntax.in.SCDocs._.SuperCollider.3.14.0-dev.Help.with.custom.css.pdf](https://github.com/supercollider/supercollider/files/15011668/HTML.LaTeX.and.MathML.Syntax.in.SCDocs._.SuperCollider.3.14.0-dev.Help.with.custom.css.pdf)
- [HTML.LaTeX.and.MathML.Syntax.in.SCDocs._.SuperCollider.3.14.0-dev.Help.with.default.css.pdf](https://github.com/supercollider/supercollider/files/15011671/HTML.LaTeX.and.MathML.Syntax.in.SCDocs._.SuperCollider.3.14.0-dev.Help.with.default.css.pdf)

**Comment by jrsurge** [comment]
Thanks for this - I can appreciate how much work has gone into this.

Unfortunately, apart from the concerns already raised, this PR is too big to review - I do not think it can be accepted in its current form.

If this is something you would like to continue with, can you please break it up into smaller, functional PRs, that each add one change?

I understand that this is a lot of work, but it takes as much work to review it effectively, and it is asking a lot of others working on the Project.

If you would like to discuss breaking it up, please let me know - but there is a question that needs to be answered first about whether this is the right direction for the Project, and I'm not sure at this point. I am also more than happy to discuss this as well, so please feel free to contact me if I can help.

**Comment by prko** [comment]
@jrsurge 

> If this is something you would like to continue with, can you please break it up into smaller, functional PRs, that each add one change?

OK, I can do that. Would the following be OK?
- subsubsection (soft-coded)
- LaTeX and MathML support
- HTML tags support (br, table, hr, audio, video etc)
Or should it be divided into smaller parts?
 
> but there is a question that needs to be answered first about whether this is the right direction for the Project, and I'm not sure at this point.

My general thought: 
The new features are necessary, even though the ways of implementation are not always ideal.
- I think the integration of LaTeX (and MathML for mathematical expression in teletype text) is very useful. The implementation seems to be OK, but many of the uploaded files for offline access to the MathJax script might be unnecessary.
- The implementation of a subsubsection is also needed to structure large documentation. It should, of course, be hard coded. However, I do not understand C++, and I already made an issue with it several years ago, but there has been no change. For now, and with my current coding skills, soft-coded implementation is the best.
- HTML tags support is not good because it should ideally be implemented as SCDoc syntax. However, the SCDoc syntax is hard coded. I cannot do it... From the view of integrating LaTeX, mixing HTML syntax is not bad. For now, and with my current coding skills, soft-coded implementation is the best.

**Comment by prko** [comment]
I close this, and will make separate PRs.

**Pull request closed without merging**